File: tests/07_phases_ompss.dg/fortran/success_task_15.f90
! <testinfo>
! test_generator=config/mercurium-ompss
! </testinfo>

 program main
 interface
 !$omp task inout(a)
 subroutine sub(a)
         character (len=80) :: a(1:20)
 end subroutine sub
 !$omp task in(a)
 subroutine sub2(a)
         character (len=80) :: a(1:20)
 end subroutine sub2
 end interface

 character (len=80) :: x(1:20)

 x(1) = 'ABCDE'
 x(2) = 'FGHIJ'
 call sub(x)

 call sub2(x)

 !$omp taskwait

 end program main

 subroutine sub(a)
         character (len=80) :: a(1:20)

         a(1) = 'KLMNO'
         a(2) = 'PQRST'
 end subroutine sub

 subroutine sub2(a)
         character (len=80) :: a(1:20)

         IF (a(1) /= 'KLMNO') STOP 1
         IF (a(2) /= 'PQRST') STOP 2
 end subroutine sub2

File: tests/07_phases_ompss.dg/c/success_task_15.c
/*--------------------------------------------------------------------
  (C) Copyright 2006-2012 Barcelona Supercomputing Center
                          Centro Nacional de Supercomputacion
  
  This file is part of Mercurium C/C++ source-to-source compiler.
  
  See AUTHORS file in the top level directory for information
  regarding developers and contributors.
  
  This library is free software; you can redistribute it and/or
  modify it under the terms of the GNU Lesser General Public
  License as published by the Free Software Foundation; either
  version 3 of the License, or (at your option) any later version.
  
  Mercurium C/C++ source-to-source compiler is distributed in the hope
  that it will be useful, but WITHOUT ANY WARRANTY; without even the
  implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
  PURPOSE.  See the GNU Lesser General Public License for more
  details.
  
  You should have received a copy of the GNU Lesser General Public
  License along with Mercurium C/C++ source-to-source compiler; if
  not, write to the Free Software Foundation, Inc., 675 Mass Ave,
  Cambridge, MA 02139, USA.
--------------------------------------------------------------------*/


/*
<testinfo>
test_generator="config/mercurium-ompss"
test_compile_fail_nanos6_mercurium=yes
test_compile_fail_nanos6_imcc=yes
</testinfo>
*/
#include<assert.h>

const int n = 1000;
int main()
{
    int x1 = -1;
    int x2 = -1;

    #pragma omp task shared(x1, x2)
    {
        #pragma omp for lastprivate(x1)
        for (x1 = 0; x1 < n; ++x1)
        {
        }

        #pragma omp for
        for (x2 = 0; x2 < n; ++x2)
        {
        }
    }

    #pragma omp taskwait
    assert(x1 == n);
    assert(x2 == -1);

    return 0;
}
